### DISCUSSION TITLE
Custom Snippets for Django Html Not Working

### DISCUSSION THREAD

**Discussion by Zahedcse** [pulsar-edit/.github#136] [Core Application & Packages]
I wrote snipppets for HTML-> django project:
'text.html.django':
  'autoescape':
    'prefix': 'autoescape'
    'body': '{% autoescape ${1:off} %}\n\t$2\n{% endautoescape %}'

  'block':
    'prefix': 'block'
    'body': '{% block ${1:block_name} %}\n\t$2\n{% endblock %}'

  'csrf':
    'prefix': 'csrf'
    'body': '{% csrf_token %}'

  'extends':
    'prefix': 'extends'
    'body': '{% extends "${1:base_template}" %}'

  'for':
    'prefix': 'for'
    'body': '{% for ${1:item} in ${2:sequence} %}\n\t$3\n{% endfor %}'

  'if':
    'prefix': 'if'
    'body': '{% if ${1:condition} %}\n\t$2\n{% endif %}'

  'ifelse':
    'prefix': 'ifelse'
    'body': '{% if ${1:condition} %}\n\t$2\n{% else %}\n\t$3\n{% endif %}'

  'include':
    'prefix': 'include'
    'body': '{% include "${1:template_name}.html" %}'

  'url':
    'prefix': 'url'
    'body': '{% url "${1:view_name}" ${2:arg1,arg2} %}'

but even after  restarting the pulsar app:
It didn't work

**Comment by savetheclocktower** [comment]
The very first line should be `'.text.html.django'` — note the dot before `text`. That'll probably fix it, but let me know if it doesn't.

**Reply by confused-Techie** [reply]
Wow, looks like we were typing at the same time lol

**Comment by confused-Techie** [comment]
The first thing that stands out to me, you've used `text.html.django`. 

While I'm not sure if what file you are working with is correctly being identified with `django`, since I don't believe Pulsar will do that natively (But might be wrong, or you might need a community package).

But this should be prefixed with a `.` like `.text.html.django`. According to our docs:

`The outermost keys are the selectors where these snippets should be active, prefixed with a period (.) (details below).`

Feel free to read them more on the [snippets](https://github.com/pulsar-edit/snippets) package.

Hopefully this helps. Otherwise opening a proper issue on the `snippets` package is probably a good idea.

**Reply by savetheclocktower** [reply]
Looks like [language-django](https://web.pulsar-edit.dev/packages/language-django) provides the `text.html.django` grammar, so I bet he's all set there.
